Configure

```diff
@@ -1088,25 +1103,18 @@ if (!$disabled{dso} && $target{dso_scheme} ne "")
 $config{ex_libs}="$libs$config{ex_libs}" if ($libs ne "");
 
 # If threads aren't disabled, check how possible they are
-unless ($disabled{threads}) {
-    if ($auto_threads) {
-        # Enabled by default, disable it forcibly if unavailable
-        if ($target{thread_scheme} eq "(unknown)") {
-            $disabled{threads} = "unavailable";
-        }
-    } else {
-        # The user chose to enable threads explicitly, let's see
-        # if there's a chance that's possible
-        if ($target{thread_scheme} eq "(unknown)") {
-            # If the user asked for "threads" and we don't have internal
-            # knowledge how to do it, [s]he is expected to provide any
-            # system-dependent compiler options that are necessary.  We
-            # can't truly check that the given options are correct, but
-            # we expect the user to know what [s]He is doing.
-            if ($no_user_cflags && $no_user_defines) {
-                die "You asked for multi-threading support, but didn't\n"
-                    ,"provide any system-specific compiler options\n";
-            }
+unless ($disabled{threads} || $auto{threads}) {
+    # The user chose to enable threads explicitly, let's see
+    # if there's a chance that's possible
+    if ($target{thread_scheme} eq "(unknown)") {
+        # If the user asked for "threads" and we don't have internal
+        # knowledge how to do it, [s]he is expected to provide any
+        # system-dependent compiler options that are necessary.  We
+        # can't truly check that the given options are correct, but
+        # we expect the user to know what [s]He is doing.
+        if ($user_cflags ne "" && !@user_defines) {
+            die "You asked for multi-threading support, but didn't\n"
+                ,"provide any system-specific compiler options\n";
```

Comment:
So that attempt to `./Configure gcc enable-threads` should terminate with this `die` message, right? It doesn't. It should be `$user_cflags eq ""`.

There is "[s]He" typo, i.e. capital H.

[On side note I find above placement of comma in die counter-intuitive. Indentation gives clear enough clue.]

Reply:
On second *side* note/thought I even wonder if it has to be that way? I mean do we actually want to provide this fall-back? I mean expecting that user specifying enable-threads would provide thread flags on config command line? Especially if we take into consideration that nowadays enable-threads kind of implies that you need scheme, and you can't do that on config command line. In other words shouldn't one instruct users to put together own Configuration/99-myconf.conf irregardless if they passed custom flags or not?

Reply:
> enable-threads kind of implies that you need thread_scheme

By this I mean that we no longer provide option for application to specify multi-threading callbacks, don't we? So that if user wants to implement own thread_scheme [s]he has to provide own crypto/threads_whatever.c replacement. With this in mind *maybe* thread_scheme should be name of this file... Or am I completely off?

Reply:
Actually, you're absolutely right.  This is a remainder from pre-1.1.0 times and cannot be supported as is with 1.1.0 and up.  OpenSSL-ish routines like this provided in `crypto/threads_whatever.c` are required.

A curiosity (and side note), btw...  it seems like with have screwed up with unixware.  The thread scheme there is 'uithreads', the C flag is `-Kthreads`, but we don't really have support for it.  As a matter of fact, the way `crypto/threads_whatever.c` are currently written, we assume pthreads, which I can only assume is faulty like hell...  except for `unixware-7-gcc`

Reply:
> screwed up with unixware

We should really discard it and few others [we never had hands-on experience with]. Suggested strategy [if anybody asks/cares] should be that legacy OS might as well run legacy release.

Reply:
> This is a remainder from pre-1.1.0 times and cannot be supported as is with 1.1.0 and up.

So you'll remove this or fix it up? I mean if not removed there is wrong condition to fix... [And typo].

Reply:
I intend to change this to a simple no-can-do warning.